File: chapter01/C++/MiniDuckSimulator.cpp
#include <iostream>
#include "Duck.hpp"
#include "MallardDuck.hpp"
#include "ModelDuck.hpp"

using namespace std;

int main()
{
    Duck *mallard = new MallardDuck;
    mallard->display();
    mallard->swim();
    mallard->performFly();
    Duck *model = new ModelDuck;
    model->performFly();
    FlyRocketPowered fb;
    model->setFlyBehavior(fb);
    model->performFly();
    /*
mallard.performQuack();
mallard.performFly();
 */
    return 0;
}

File: chapter01/java/MiniDuckSimulator.java

public class MiniDuckSimulator {

    public static void main(String[] args) {
        Duck mallard = new MallardDuck();
        mallard.display();
        mallard.swim();
        mallard.performFly();
        Duck model = new ModelDuck();
        model.performFly();
        model.setFlyBehavior(new FlyRocketPowered());
        model.performFly();
        /*
 mallard.performQuack(); 
 model.performFly();  */
    }
}
